src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -189,7 +189,7 @@ public AuthorizeFilter(string policy)
             var authenticateResult = await policyEvaluator.AuthenticateAsync(effectivePolicy, context.HttpContext);
 
             // Allow Anonymous skips all authorization
-            if (context.Filters.Any(item => item is IAllowAnonymousFilter))
+            if (HasAllowAnonymous(context.Filters))
```

Comment:
To confirm:
- The later `ToArray()` calls are the only thing preventing removal of the Linq dependency?
- This change is just for perf and consistency and no behaviours change?

Reply:
> This change is just for perf and consistency and no behaviours change?

Yup. I noticed this unnecessary Linq use when I was making a similar change to the Antiforgery filter.